addons/product/models/product_pricelist.py

```diff
@@ -307,10 +307,26 @@ def _get_partner_pricelist_multi(self, partner_ids):
         pl_domain = self._get_partner_pricelist_multi_search_domain_hook(company_id)
 
         # if no specific property, try to find a fitting pricelist
-        result = Property._get_multi('property_product_pricelist', Partner._name, partner_ids)
+        specific_properties = Property._get_multi(
+            'property_product_pricelist', Partner._name,
+            list(models.origin_ids(partner_ids)),  # Some NewID can be in the partner_ids
+        )
+        result = {}
+        remaining_partner_ids = []
+        for pid in partner_ids:
+            if (
+                specific_properties.get(pid) and
+                specific_properties[pid]._get_partner_pricelist_multi_filter_hook()
```

Comment:
```suggestion
                specific_properties.get(pid)
                and specific_properties[pid]._get_partner_pricelist_multi_filter_hook()
```
Personal opinion, but I find it way clearer to have the operator on the next line, to clearly see how the next line is considered in the condition.